ModuleFifthTasks/Task1/LinqTask.cs

```diff
@@ -8,63 +9,77 @@ public static class LinqTask
     {
         public static IEnumerable<Customer> Linq1(IEnumerable<Customer> customers, decimal limit)
         {
-            throw new NotImplementedException();
+            return customers.Where(customer => customer.Orders.Sum(order => order.Total) > limit);
         }
 
         public static IEnumerable<(Customer customer, IEnumerable<Supplier> suppliers)> Linq2(
             IEnumerable<Customer> customers,
             IEnumerable<Supplier> suppliers
         )
         {
-            throw new NotImplementedException();
+            return customers.Select((customer, num) => new ValueTuple<Customer, IEnumerable<Supplier>>(customer, suppliers.Where(supplier => supplier.Country == customer.Country && supplier.City == customer.City)));
         }
 
         public static IEnumerable<(Customer customer, IEnumerable<Supplier> suppliers)> Linq2UsingGroup(
             IEnumerable<Customer> customers,
             IEnumerable<Supplier> suppliers
         )
         {
-            throw new NotImplementedException();
+            return customers.GroupJoin(
+                suppliers,
+                customer => customer,
+                supplier => new Customer(),
+                (c, s) => new ValueTuple<Customer, IEnumerable<Supplier>>(c, s.Where(supplier => supplier.Country == c.Country && supplier.City == c.City)));
         }
 
         public static IEnumerable<Customer> Linq3(IEnumerable<Customer> customers, decimal limit)
         {
-            throw new NotImplementedException();
+            return customers.Where(customer => customer.Orders.Any(order => order.Total >= limit));
         }
 
         public static IEnumerable<(Customer customer, DateTime dateOfEntry)> Linq4(
             IEnumerable<Customer> customers
         )
         {
-            throw new NotImplementedException();
+            return customers.Where(customer => customer.Orders.Count() > 0).Select(customer => new ValueTuple<Customer, DateTime>(customer, customer.Orders.Min(order => order.OrderDate)));
```

Comment:
Do not forget to use line breaks 😃 Also **please note** that `.Any()` is more effective than `.Count() > 0` becase it will stop iteration over collection after the first element.

```suggestion
           return customers
                .Where(customer => customer.Orders.Any())
                .Select(customer => (customer, customer.Orders.Min(order => order.OrderDate)));
```

Reply:
Fixed in d71bf5c.